src/rpp/rpp/schedulers/trampoline_scheduler.hpp

```diff
@@ -96,22 +121,22 @@ class trampoline final : public details::scheduler_tag
         rpp::composite_subscription m_sub;
     };
 
-    static auto create_worker(const rpp::composite_subscription& sub = composite_subscription{})
+    /**
+     * \brief Take ownership over queue and return handle which would drain queue during destruction
+     */
+    [[nodiscard]] static utils::finally_action<void(*)()> ensure_queue_if_no_any_owner()
     {
-        return worker<worker_strategy>{sub};
+        auto& queue = get_schedulable_queue();
+        if (queue.has_value())
+            return utils::finally_action{+[] {}};
+
+        queue = std::priority_queue<current_thread_schedulable>{};
+        return utils::finally_action{&drain_queue};
     }
```

Comment:
nit: For better readability, would you mind to move this function to closer to `wait_and_extract_executable_if_subscribed` at top?

Reply:
nvm, it's a **public** static method.